pkg/egressgateway/policy.go

```diff
@@ -10,15 +10,48 @@ import (
 	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/types"
 
+	"github.com/cilium/cilium/pkg/datapath/linux/route"
 	k8sConst "github.com/cilium/cilium/pkg/k8s/apis/cilium.io"
 	v2alpha1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
 	k8sLabels "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/labels"
 	slim_metav1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
 	"github.com/cilium/cilium/pkg/logging/logfields"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
 	"github.com/cilium/cilium/pkg/policy"
 	"github.com/cilium/cilium/pkg/policy/api"
 )
 
+// policyGatewayConfig is the internal representation of an egress gateway,
+// describing which node should act as egress gateway for a given policy.
+type policyGatewayConfig struct {
+	nodeSelector api.EndpointSelector
+	iface        string
+	egressIP     net.IP
+}
+
+// gatewayConfig is the gateway configuration derived at runtime from a policy.
+//
+// Some of these fields are derived from the running system as the policy may
+// specify only the egress IP (and so we need to figure out which interface has
+// that IP assigned to) or the interface (and in this case we need to find the
+// first IPv4 assigned to that).
```

Comment:
I think we might want to point this out in the docs as well - any subsequent change to a node's egress interface config won't be reconciled with the policy.